Content.Shared/Actions/SharedActionsSystem.cs

```diff
@@ -299,7 +307,7 @@ private void OnActionRequest(RequestPerformActionEvent ev, EntitySessionEventArg
             performEvent.Performer = user;
 
         // All checks passed. Perform the action!
-        PerformAction(user, component, actionEnt, action, performEvent, curTime);
+        PerformAction(action.PreferredEntity, component, actionEnt, action, performEvent, curTime);
```

Comment:
just make the event target a separate argument,, the "performer"/user should always be the entity that owns the `ActionsComponent`

Reply:
Then this makes a triple check in that method between the performer because the check right now checks the performer (this) but then that method checks if the performer is null it runs on the container.